app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,26 @@
+class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def google_oauth2
+    @user = User.from_omniauth(request.env["omniauth.auth"])
+
+    if @user.persisted?
+      flash[:notice] = "devise.omniauth_callbacks.success", :kind => "Google"
```

Comment:
This use of the flash doesn't look right. You're just passing in a string literal with the value "devise.omniauth_callbacks.success", and I don't know how the portion after the comma will be interpreted. I believe the syntax you're looking for is:

``I18n.t('devise.omniauth_callbacks.success', :kind => 'Facebook')

That should pull the right value from the Devise localization file (config/locales/devise.en.yml) using the Rails I18n gem: http://guides.rubyonrails.org/i18n.html